common/app-starters-postprocessor-common/src/main/java/org/springframework/cloud/stream/app/postprocessor/ContentTypeEnvironmentPostProcessor.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2018 the original author or authors.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.stream.app.postprocessor;
+
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.env.EnvironmentPostProcessor;
+import org.springframework.cloud.stream.messaging.Sink;
+import org.springframework.cloud.stream.messaging.Source;
+import org.springframework.core.env.ConfigurableEnvironment;
+import org.springframework.core.env.PropertiesPropertySource;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+
+/**
+ * An {@link EnvironmentPostProcessor} to set the {@code spring.cloud.stream.bindings.{input,output}.contentType}
+ * channel properties to a default of {@code application/octet-stream} if it has not been set already.
+ *
+ * Subclasses may extend this class to change the default content type and channel name(s).
+ *
+ * @author Chris Schaefer
+ */
+public class ContentTypeEnvironmentPostProcessor implements EnvironmentPostProcessor {
+	private Map<String, String> channelMap = createChannelMap();
+
+	private Map<String, String> createChannelMap() {
+		Map<String, String> channelMap = new HashMap<>();
+		channelMap.put(Sink.INPUT, "application/octet-stream");
+		channelMap.put(Source.OUTPUT, "application/octet-stream");
+
+		return channelMap;
+	}
+
+	protected static final String PROPERTY_SOURCE_KEY_NAME = ContentTypeEnvironmentPostProcessor.class.getName();
+	protected static final String CONTENT_TYPE_PROPERTY_PREFIX = "spring.cloud.stream.bindings.";
+	protected static final String CONTENT_TYPE_PROPERTY_SUFFIX = ".contentType";
+
+	public ContentTypeEnvironmentPostProcessor() {
+		super();
+	}
+
+	protected ContentTypeEnvironmentPostProcessor(Map<String, String> channelMap) {
+		this.channelMap = channelMap;
+	}
+
+	protected ContentTypeEnvironmentPostProcessor(String channelName, String contentType) {
```

Comment:
Why do we lose the way just override the content-type? That is exactly what we will do in most cases. Just because all our app starters are based on only conventional `Sink.INPUT` and `Source.OUTPUT`.

Reply:
you can still override the content-type, your just providing an explicit channel name (figured maybe somebody would want to keep`output` as the postprocessor default and say `input` as something else).

so would you prefer this constructor taking a single parameter being `contentType` which would set all channel content types or are you looking to keep this as-is for explicit channel setting and add another constructor that only takes a`contentType` parameter and updates all channels?

Reply:
I'm looking strictly to way to override all by single ctor. In most cases we have a problem with `source` or `sink` app type, therefore we get deal only with one channel and that would be a bit confusing do not let to reconfigure just content type without an attention that there are both channels, when one of them isn't going to be used at all.

That's confusing, too, frankly, but we can reconsider that part in the future when someone will complain.

For now I'm fine with two (or more channel) to be configured.

Thanks

Reply:
wouldn't it be true that both of them may be used when marked as `@Processor`? in the docs it states "Processor can be used for an application which has both an inbound channel and an outbound channel."

so currently we have:
* map ctor - handle multiple mappings
* channelName, contentType - handle one mapping (seemed handy in processor case if you just want to change one channel)

i can drop `channelName, contentType` and replace it with `contentType` if you prefer, just wanted to mention the `@Processor` docs. if i am on the right track in thoughts about `@Processor` would it be worthwhile to keep `channelName, contentType` and add a blanket `contentType` override so you have the ability to cover (seemingly) all cases?

Reply:
You are absolutely on the same page as me! 😄 
Only what I want to see here is one more conventional `ContentTypeEnvironmentPostProcessor(String contentType)`  to override all the mapping because exactly this one will be used in most cases in sources or sinks.

Reply:
okay got it - latest push should do it